### PR TITLE
[Core] Make JobSupervisor logs structured

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
`JobSupervisor` is a system actor and this PR makes its logs structured.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/ray_constants.py [hunk 1]
```diff
@@ -500,9 +500,10 @@ def gcs_actor_scheduling_enabled():
...
 RAY_TPU_MAX_CONCURRENT_CONNECTIONS_ENV_VAR = "RAY_TPU_MAX_CONCURRENT_ACTIVE_CONNECTIONS"
 
-
 RAY_NODE_IP_FILENAME = "node_ip_address.json"
 
 PLACEMENT_GROUP_BUNDLE_RESOURCE_NAME = "bundle"
 
 RAY_LOGGING_CONFIG_ENCODING = os.environ.get("RAY_LOGGING_CONFIG_ENCODING")
+
+RAY_BACKEND_LOG_JSON_ENV_VAR = "RAY_BACKEND_LOG_JSON"
```
Comment (kevin85421) [inline python/ray/_private/ray_constants.py:509]: The env name is a bit vague.

File: python/ray/dashboard/modules/job/job_supervisor.py [hunk 2]
```diff
@@ -101,8 +103,17 @@ def _configure_logger(self) -> None:
             f"jobs/supervisor-{self._job_id}.log",
         )
         os.makedirs(os.path.dirname(supervisor_log_file_name), exist_ok=True)
-        self._logger.addHandler(logging.StreamHandler())
-        self._logger.addHandler(logging.FileHandler(supervisor_log_file_name))
+        self._logger.addFilter(CoreContextFilter())
```
Comment (kevin85421) [inline python/ray/dashboard/modules/job/job_supervisor.py:106]: My current concern is that Ray has too many loggers, and each logger is configured in a different place. If users want to enable structured logging, they need to configure multiple settings.

**Review by kevin85421 — APPROVED** [review]

File: python/ray/_private/ray_constants.py (hunk 1 above)
Comment (jjyao) [reply python/ray/_private/ray_constants.py:509]: That's an existing env var that we used: it controls all system logs.

File: python/ray/dashboard/modules/job/job_supervisor.py (hunk 2 above)
Comment (jjyao) [reply python/ray/dashboard/modules/job/job_supervisor.py:106]: There are only two settings: LoggingConfig for application logs and RAY_BACKEND_LOG_JSON for system logs (JobSupervisor is system log).

**Pull request merged**